Lib/platform.py

```diff
@@ -1230,6 +1230,65 @@ def platform(aliased=0, terse=0):
     _platform_cache[(aliased, terse)] = platform
     return platform
 
+### freedesktop.org os-release standard
+# https://www.freedesktop.org/software/systemd/man/os-release.html
+
+# NAME=value with optional quotes (' or "). The regular expression is less
+# strict than shell lexer, but that's ok.
+_os_release_line = re.compile(
+    "^(?P<name>[a-zA-Z0-9_]+)=(?P<quote>[\"\']?)(?P<value>.*)(?P=quote)$"
+)
+# /etc takes precedence over /usr/lib
+_os_release_candidates = ("/etc/os-release", "/usr/lib/os-relesase")
+_os_release_cache = None
+
+
+def _parse_os_release(lines):
+    # These fields are mandatory fields with well-known defaults
+    # in pratice all Linux distributions override NAME, ID, and PRETTY_NAME.
+    info = {
+        "NAME": "Linux",
+        "ID": "linux",
+        "PRETTY_NAME": "Linux",
+    }
+
+    for line in lines:
+        mo = _os_release_line.match(line)
+        if mo is not None:
+            info[mo.group('name')] = mo.group('value')
+
+    # ID_LIKE is a space separated field of ids
+    if 'ID_LIKE' in info:
+        info['ID_LIKE'] = tuple(
+            id_like for id_like in info['ID_LIKE'].split(' ') if id_like
+        )
```

Comment:
I suggest to remove this code.

I'm not sure if it's worth it to have a special case only for the ID_LIKE variable. The platform module is mostly based on strings. You introduce a special case where a single variable is a tuple. Since it's well specified that it's space separated, the caller can split the string by themselves, no?

Splitting at spaces it trivial in Python:

```
>>> "debian ubuntu".split()
['debian', 'ubuntu']
```

Reply:
This is the only special case that a lot of users will get wrong and that won't get detected by CI. For almost all distros the ``ID_LIKE`` is either not present or it is a single entry, often ``debian`` or ``fedora``. For example Ubuntu has ``ID_LIKE=debian``. Only very few distros have an ID_LIKE with multiple entries.

Users always have to split the field and always check for "name in info["ID_LIKE"].split()". It's a foot gun that is solved easily.

Reply:
The problem is that if tomorrow a new field is added with a value separated by spaces, the caller might expect Python to magically split it as a tuple. But now, we will have to update the function whenever a new field is added for add a special case.

If there is no special case and it's a dummy parser, the code is more likely to stay for the same for the next 10 years.

I would prefer to not reintroduce the same problem that we fixed by removing linux_distribution(): having to update the stdlib when the filename changes or file content evolves.

If you want a more advanced view of the os-release, I suggest to write a higher level API on top of it and put it on PyPI, so it can be easily updated (similar to the distro module which contains many special cases).